lib/ipmi_fru.c

```diff
@@ -3849,40 +3853,41 @@ static int ipmi_fru_upg_ekeying(struct ipmi_intf *intf, char *pFileName,
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to get multirec size from file '%s'.",
 			pFileName);
-		return -1;
+		ERR_EXIT;
 	}
 	buf = malloc(fileMultiRecSize);
 	if (!buf) {
 		lprintf(LOG_ERR, "ipmitool: malloc failure");
-		return -1;
+		ERR_EXIT;
 	}
 	if (ipmi_fru_get_multirec_from_file(pFileName, buf, fileMultiRecSize,
 					    offFileMultiRec)
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to get multirec from file '%s'.",
 			pFileName);
-		free(buf);
-		buf = NULL;
-		return -1;
+		ERR_EXIT;
 	}
 	if (ipmi_fru_get_adjust_size_from_buffer(buf, &fileMultiRecSize) != 0) {
 		lprintf(LOG_ERR, "Failed to adjust size from buffer.");
-		free(buf);
-		buf = NULL;
-		return -1;
+		ERR_EXIT;
 	}
 	if (write_fru_area(intf, &fruInfo, fruId, 0, offFruMultiRec,
 			   fileMultiRecSize, buf)
 	    != 0) {
 		lprintf(LOG_ERR, "Failed to write FRU area.");
+		ERR_EXIT;
+	}
+
+exit:
+	if (buf) {
 		free(buf);
 		buf = NULL;
-		return -1;
 	}
-	free(buf);
-	buf = NULL;
-	lprintf(LOG_INFO, "Done upgrading Ekey.");
-	return 0;
+
+	if (!rc) {
```

Comment:
Looks like it's safe to assume that anywhere above `exit:` but after the closing curly of the last `if()` is an `!rc` zone. So let's just move this message there and get rid of this check.